projects/plugins/crm/src/automation/commons/conditions/contacts/class-contact-transitional-status.php

```diff
@@ -0,0 +1,182 @@
+<?php
+/**
+ * Jetpack CRM Automation Contact_Transitional_Status condition.
+ *
+ * @package automattic/jetpack-crm
+ */
+
+namespace Automattic\Jetpack\CRM\Automation\Conditions;
+
+use Automattic\Jetpack\CRM\Automation\Automation_Exception;
+use Automattic\Jetpack\CRM\Automation\Automation_Logger;
+use Automattic\Jetpack\CRM\Automation\Base_Condition;
+
+/**
+ * Contact_Transitional_Status condition class.
+ *
+ * @since $$next-version$$
+ */
+class Contact_Transitional_Status extends Base_Condition {
+
+	/**
+	 * The Automation logger.
+	 *
+	 * @since $$next-version$$
+	 * @var Automation_Logger $logger The Automation logger.
+	 */
+	private $logger;
+
+	/**
+	 * All valid operators for this condition.
+	 *
+	 * @since $$next-version$$
+	 * @var string[] $valid_operators Valid operators.
+	 */
+	private $valid_operators = array(
+		'from_to',
+	);
+
+	/**
+	 * All valid attributes for this condition.
+	 *
+	 * @since $$next-version$$
+	 * @var string[] $valid_operators Valid attributes.
+	 */
+	private $valid_attributes = array(
+		'previous_status_was',
+		'new_status_is',
+	);
+
+	/**
+	 * Contact_Transitional_Status constructor.
+	 *
+	 * @since $$next-version$$
+	 *
+	 * @param array $step_data The step data for the condition.
+	 */
+	public function __construct( array $step_data ) {
+		parent::__construct( $step_data );
+
+		$this->logger = Automation_Logger::instance();
+	}
+
+	/**
+	 * Executes the condition. If the condition is met, the value stored in the
+	 * attribute $condition_met is set to true; otherwise, it is set to false.
+	 *
+	 * @since $$next-version$$
+	 *
+	 * @param array $data The data this condition has to evaluate.
+	 * @return void
+	 *
+	 * @throws Automation_Exception If an invalid operator is encountered.
+	 */
+	public function execute( array $data ) {
+		if ( ! $this->is_valid_contact_status_transitional_data( $data ) ) {
+			$this->logger->log( 'Invalid contact status transitional data', $data );
+			$this->condition_met = false;
+
+			return;
+		}
+
+		$operator   = $this->get_attributes()['operator'];
+		$status_was = $this->get_attributes()['previous_status_was'];
+		$status_is  = $this->get_attributes()['new_status_is'];
+
+		$this->logger->log( 'Condition: Contact_Transitional_Status ' . $operator . ' ' . $status_was . ' => ' . $status_is );
+		switch ( $operator ) {
+			case 'from_to':
+				$this->condition_met = ( $data['old_status_value'] === $status_was ) && ( $data['contact']['data']['status'] === $status_is );
+				$this->logger->log( 'Condition met?: ' . ( $this->condition_met ? 'true' : 'false' ) );
+
+				return;
+		}
+
+		$this->condition_met = false;
+		$this->logger->log( 'Invalid operator: ' . $operator );
+
+		throw new Automation_Exception( 'Invalid operator: ' . $operator );
```

Comment:
This could be a check at the start of the function like the contact status transitional data check and use `$valid_operators` to explicitly check that the operator is valid, and then have a default case in the `operator_switch` which logs something saying that we have a valid but unimplemented operator.

Reply:
Thanks for the suggestion! Added in 594db33.